SoObjects/Mailer/SOGoMailFolder.m

```diff
@@ -104,10 +104,11 @@ static NSComparisonResult _compareFetchResultsByMODSEQ (id entry1, id entry2, vo
   return [modseq1 compare: modseq2];
 }
 
-static NSInteger _compareFetchResultsByUID (id entry1, id entry2, NSDictionary *uids)
+static NSComparisonResult _compareFetchResultsByUID (id entry1, id entry2, void *uidsp)
```

Comment:
Not sure to understand the utility the void* then the NSDictionnary* casting? Can you check again if this causes a fail?

Reply:
Without this:

```
SOGoMailFolder.m:233:52: error: incompatible function pointer types sending 'NSInteger (id, id, NSDictionary *)' (aka 'long (id, id, NSDictionary *)') to parameter of type 'NSComparisonResult (*)(id, id, void *)' (aka 'enum NSComparisonResult (*)(id, id, void *)') [-Wincompatible-function-pointer-types]
  233 |   result = [fetchResults sortedArrayUsingFunction: _compareFetchResultsByUID
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/Foundation/NSArray.h:145:44: note: passing argument to parameter 'comparator' here
  145 |     (NSComparisonResult (*)(id, id, void*))comparator
      |                                            ^
```